target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -394,4 +394,88 @@ abstract contract Pyth is
     function version() public pure returns (string memory) {
         return "1.4.3";
     }
+
+    // TWAP verification and update methods
+    function updateTwapFeed(
+        bytes[] calldata updateData,
+        bytes32 feedId,
+        uint64 startTime,
+        uint64 endTime
+    ) external payable {
+        // Parse and validate both start and end updates
+        PythStructs.PriceFeed[] memory startFeed = parsePriceFeedUpdatesInternal(
```

Comment:
the messages here have a different format. They are of type TwapMessage -- see the file pythnet/pythnet_sdk/src/messages.rs for the spec. You will have to implement a different parsing function to read this data